### PR TITLE
#9Task. AleksanderGogolushko

### PR DESCRIPTION
**Pull request by AleksandrGogolushko**
https://travis-ci.org/github/AleksandrGogolushko/js-assignments/builds/707125712

### CODE REVIEW

File: task/08-objects-tasks.js [hunk 1]
```diff
@@ -108,36 +108,41 @@ function fromJSON(proto, json) {
...
+    combine: function (selector1, combinator, selector2) {
+        throw new Error('Not implemented');
+    },
+    stringify: function () {
         throw new Error('Not implemented');
     },
+    validation: function (sequince) {
+        throw new Error('Not implemented');
+    }
 };
```
Comment (ilya4321) [inline task/08-objects-tasks.js]: changes from task8 are noise for the PR. Could you remove these please?

File: task/09-functions-n-closures-tasks.js [hunk 2]
```diff
@@ -65,10 +66,16 @@ function getPowerFunction(exponent) {
  *   getPolynom()      => null
  */
 function getPolynom() {
-    throw new Error('Not implemented');
+    let arg = arguments;
```
Comment (ilya4321) [inline task/09-functions-n-closures-tasks.js]: "arguments" is deprecated for latest versions of JS. You can use 'rest parameters' instead

**Review by IlyaMokin — COMMENTED** [review]
I'll mark the PR as draft, when it will be ready please click "ready for review"
Thank you!

File: task/09-functions-n-closures-tasks.js [hunk 3]
```diff
@@ -104,10 +112,17 @@ function memoize(func) {
  * retryer() => 2
  */
 function retry(func, attempts) {
-    throw new Error('Not implemented');
+    return () => {
+        for (let i = 0; i < attempts; i++)
```
Comment (ilya4321) [inline task/09-functions-n-closures-tasks.js]: I've noticed that you don't like brackets.
There are a lot of examples when[ the approach creates](https://stackoverflow.com/questions/2125066/is-it-a-bad-practice-to-use-an-if-statement-without-curly-braces) problems. So general rule is always use brackets.

File: task/09-functions-n-closures-tasks.js [hunk 4]
```diff
@@ -64,11 +65,17 @@ function getPowerFunction(exponent) {
  *   getPolynom(8)     => y = 8
  *   getPolynom()      => null
  */
-function getPolynom() {
-    throw new Error('Not implemented');
+function getPolynom(...args) {
+    let arg = args
```
Comment (ilya4321) [inline task/09-functions-n-closures-tasks.js]: it looks like `arg` is redundant here.

File: task/09-functions-n-closures-tasks.js [hunk 5]
```diff
@@ -132,7 +148,13 @@ function retry(func, attempts) {
  *
  */
 function logger(func, logFunc) {
-    throw new Error('Not implemented');
+    return function () {
+        let arg = Array.from(arguments).map(e => JSON.stringify(e)).join()
```
Comment (ilya4321) [inline task/09-functions-n-closures-tasks.js]: with `rest parameters` you even don't need the `Array.from`. The same for functions bellow

File: task/09-functions-n-closures-tasks.js [hunk 6]
```diff
@@ -150,7 +171,11 @@ function logger(func, logFunc) {
  *   partialUsingArguments(fn, 'a','b','c','d')() => 'abcd'
  */
 function partialUsingArguments(fn) {
-    throw new Error('Not implemented');
+    let arg = Array.from(arguments).filter((e, i) => i != 0)
```
Comment (ilya4321) [inline task/09-functions-n-closures-tasks.js]: most interesting approach with `rest parameters` can be used here to avoid  `filter`

**Comment by IlyaMokin** [comment]
Closing as accepted!

**Pull request closed without merging**